store/postgres_graphql.go

```diff
@@ -389,6 +398,80 @@ func psqlSubQuery(tenant string, graph *schemaGraph, qb *sqlQueryBuilder, root *
 		}
 	}
 
+	// Process distinct_on, if available. At this point all table aliases are known.
+	if distinctOnArg != nil {
+
+		// This argument may only be used, if the `order_by` argument is also used.
+		if orderByArg == nil {
+			return fmt.Errorf("distinct_on argument cannot be used without matching order_by argument")
+		}
+
+		arg := distinctOnArg
+
+		// TODO validate impedance matching between the `distinct_on` and `order_by` argument values
+
+		// This argument's value is a slice of values representing (table, field) tuples.
+		distinctOnItems, ok := (arg.Value.GetValue()).([]ast.Value)
+		if !ok {
+			return fmt.Errorf("wrong argument type in `distinct_on`: %s: %#v", arg.Name.Value, arg.Value.GetValue())
+		}
+
+		distinctOnAliases := []string{}
+
+		// Process each of the (table, field) objects
+		for _, item := range distinctOnItems {
+
+			fields, ok := item.GetValue().([]*ast.ObjectField)
+			if !ok {
+				return fmt.Errorf("failed to type cast argument value: %#v", item.GetValue())
+			}
+
+			// FIXME probably best to make it a proper type {table, field}
+			v := map[string]string{}
```

Comment:
Maybe all the parsing and creation of `v` could be separated into a function? As it's not really relevant to the rest of the function and adds complexity.

Reply:
Maybe! I now have another implementation of `order_by` and `distinct_on` (derived from this one, nevertheless) which factors out the common functionality between `order_by` and `distinct_on`, and  is using functions to split work into smaller chunks. 

The thing is, I'm not convinced that that is more readable because function calls add an extra layer of abstraction to parse visually, but I've also added more comments, so it's hard to judge. 

I'll make another PR so that two approaches can be compared and contrasted and we can make an informed decision which approach is more suited for the future.

Reply:
The PR I am referencing above is #92